src/AutoRest.CSharp/Common/Output/Builders/MultipartSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,265 @@
+                {
+                    contentTypeExpression = Literal(valueSerialization.ContentType);
+                }
+            }
+            if (valueSerialization.Type.IsFrameworkType && valueSerialization.Type.FrameworkType == typeof(BinaryData))
+            {
+                contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+                if (valueSerialization.FileName != null)
+                {
+                    fileNameExpression = Literal(valueSerialization.FileName);
+                } else
+                {
+                    fileNameExpression = name.Add(Literal(".wav"));
+                }
+                if (valueSerialization.ContentType != "application/json")
+                {
+                    contentTypeExpression = Literal(valueSerialization.ContentType);
+                }
+            }
+            if (valueSerialization.Type.IsFrameworkType)
+            {
+                contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+            }
+            if (contentExpression != null)
+            {
+                return Configuration.ApiTypes.GetMultipartFormDataRequestContentAddStatment(mulitpartContent, contentExpression, name, fileNameExpression, contentTypeExpression);
+            }
+
+            switch (valueSerialization.Type.Implementation)
+            {
+                case SerializableObjectType serializableObjectType:
+                    contentExpression = BuildValueSerizationExpression(serializableObjectType.Type, valueExpression);
+                    break;
+                case ObjectType:
+                    contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+                    break;
+                case EnumType { IsIntValueType: true, IsExtensible: false } enumType:
+                    contentExpression = BuildValueSerizationExpression(typeof(int), new CastExpression(valueExpression, enumType.ValueType));
+                    break;
+                case EnumType { IsNumericValueType: true } enumType:
+                    contentExpression = BuildValueSerizationExpression(typeof(float), new EnumExpression(enumType, valueExpression).ToSerial());
+                    break;
+                case EnumType enumType:
+                    contentExpression = BuildValueSerizationExpression(typeof(string), new EnumExpression(enumType, valueExpression).ToSerial());
+                    break;
+                default:
+                    throw new NotSupportedException($"Cannot build serialization expression for type {valueSerialization.Type}, please add `CodeGenMemberSerializationHooks` to specify the serialization of this type with the customized property");
+            }
+            return Configuration.ApiTypes.GetMultipartFormDataRequestContentAddStatment(mulitpartContent, contentExpression, name, fileNameExpression, contentTypeExpression);
```

Comment:
These statements/expression related content should go to the `ExtensibleSnippets` class: https://github.com/Azure/autorest.csharp/blob/13aaeb8b85373ea1a9074fff8e54458cec0d7c3e/src/AutoRest.CSharp/Common/Output/Expressions/ExtensibleSnippets.cs#L9